src/core/logical_layers/rfc.md

```diff
@@ -0,0 +1,73 @@
+# Layer types
+
+Before answering the question what are logical layers, we should first consider what kind of layers we have in the context of a map.
+
+## Map Style Layers
+
+The [Style Specification](https://maplibre.org/maplibre-gl-js-docs/style-spec/) defines a map layer as a way of representing data.
+Such a layer is part of the map style and can be of type "fill", "line", "symbol", "circle" and so on.
+In the context of our application, we refer to such layers as map style layers, or just a style layer
+
+## Sources of layers
+
+If we divide layers by data delivery format, we can identify "vector", "raster" and "geojson" layers.
+However, this is a simplification - despite the fact that it directly affects the way the layer style is described, this division still refers to _layer sources_.
+At the same time, the vector data source type also has layers, which we call source-layer
+
+## Logical layers
+
+And, at the top of the list are the logical layers. A logical layer is a concatenation of several data and style layers linked together by a common theme.
+
+This is how these layers look like a tree:
+
+- logical layer
+  - style layers
+  - layer source
+    - source layer
```

Comment:
```suggestion
- logical layer
  - style layer #1
  - layer #1 sources 
    - source layer
  - style layer #2
  - layer #2 sources
    - source layer (a)
    - source layer (b)
```
couple of suggestions
also, is that logic applicable?

Reply:
If we go into such detail, we are misleading that only such an irrationality can be as in the example above.
In fact - the relationship of style layer - source layer - many to many, and every logical layer can have many style layers, but only one source layer. And it's not a fundamental limitation, it's just the way it's done now, and it may change as soon as we have that need, so I'm not focusing on that here

Reply:
but not in a tree structure

Reply:
your choise

Reply:
real world tree added